kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java

```diff
@@ -149,6 +143,12 @@ public String getActiveProcessImage(String containerId, long procInstId, String
                 populateSubProcessLink(containerId, completeNode, subProcessLinks);
             }
 
+            for (NodeInstanceDesc activeNode : activeLogs) {
+                active.put(activeNode.getId(), activeNode.getNodeId());
```

Comment:
With the move of this foreach cycle we end up in reversed order of operations.
Previously:
1. active.put (all ever activated nodes)
2. active.remove (those nodes already completed)

After this change:
1. active.remove over an empty map (no-op)
2. active.put (all ever activated nodes)

Doesn't this mean that `active` now contains also those that are already completed (cause we "skipped" the remove operation)?

@nmirasch wdyt?

Reply:
I think these codes want to fix https://issues.redhat.com/browse/JBPM-5304.

Reply:
updated